internal/db/resource.go

```diff
@@ -106,3 +111,65 @@ func (d *Database) ListResources(ctx context.Context, limit, offset int) (_ []Re
 	}
 	return resources, nil
 }
+
+// buildQuery is a utility function to build the database query according to two optional parameters.
+// namePrefixFilter: used to match resources name prefix.
+// typeFilter: used to match resources type. If this is not empty the resources are fetched from a single table.
+func buildQuery(db *gorm.DB, offset, limit int, namePrefixFilter, typeFilter string) (*gorm.DB, error) {
+	// if namePrefixEmpty set to true we have a 'TRUE IS TRUE' in the SQL WHERE statement, which disable the filtering.
+	namePrefixEmpty := namePrefixFilter == ""
+	namePrefixFilter += "%"
+
+	applicationOffersQuery := db.Select(selectApplicationOffers).
+		Model(&dbmodel.ApplicationOffer{}).
+		Where("? IS TRUE OR application_offers.name LIKE ?", namePrefixEmpty, namePrefixFilter).
```

Comment:
Thanks for applying this suggestion, but I felt a bit suspicious about PostgreSQL expression evaluation (and short-circuiting of condition), and found out there is no such guaranty in PostgreSQL (See [here](https://www.postgresql.org/docs/14/sql-expressions.html#SYNTAX-EXPRESS-EVAL)). The gist is we have to use the `CASE` statement. With that we don't need the extra boolean argument:

```sql
WHERE (CASE WHEN ? = '' THEN TRUE ELSE application_offers.name LIKE ? END)
```

I tried this with PostgreSQL and it seems working.

Reply:
i was not aware of that.
Updated